lib/Conversion/LWEToPolynomial/LWEToPolynomial.cpp

```diff
@@ -116,6 +124,176 @@ struct ConvertRLWEDecrypt : public OpConversionPattern<RLWEDecryptOp> {
   }
 };
 
+struct ConvertRLWEEncrypt : public OpConversionPattern<RLWEEncryptOp> {
+  ConvertRLWEEncrypt(mlir::MLIRContext *context)
+      : OpConversionPattern<RLWEEncryptOp>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      RLWEEncryptOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    auto loc = op.getLoc();
+    auto input = adaptor.getInput();
+    auto key = adaptor.getKey();
+
+    auto inputT = cast<lwe::RLWEPlaintextType>(op.getInput().getType());
+    auto inputEncoding = inputT.getEncoding();
+    auto cleartextBitwidth =
```

Comment:
As we discussed in the past, the cleartext width/start approach might not be sufficient to express, e.g., CKKS

Reply:
BTW: drafting the new LWE types here, it would suffice for me if we could add a TODO comment saying to migrate to plaintext modulus and I'm happy to rebase my PR over it once we finish reviewing it